src/main/scala/org/constellation/storage/CheckpointService.scala

```diff
@@ -135,6 +138,79 @@ class CheckpointService[F[_]: Sync: LiftIO](
       (s: String) => ???
     )
 
+  def accept(checkpoint: FinishedCheckpoint)(implicit dao: DAO): F[Unit] = {
+
+    val obtainPeers = dao.readyPeers.map { ready =>
+      val filtered = ready.filter(t => checkpoint.facilitators.contains(t._1))
+      (if (filtered.isEmpty) ready else filtered)
+        .map(p => PeerApiClient(p._1, p._2.client))
+        .toList
+    }
+
+    (dao.nodeState, checkpoint.checkpointCacheData.checkpointBlock) match {
+      case (_, None) => Sync[F].raiseError(MissingCheckpointBlockException)
+      case (NodeState.Ready, Some(cb)) =>
+        val acceptance = for {
+          _ <- syncPending(pendingAcceptanceFromOthers, cb.baseHash)
+          _ <- Sync[F].delay { logger.debug(s"[${dao.id.short}] starting accept block: ${cb.baseHash} from others") }
+          peers <- LiftIO[F].liftIO(obtainPeers)
+          _ <- resolveMissingParents(cb, peers)
+          _ <- accept(checkpoint.checkpointCacheData)
+          _ <- pendingAcceptanceFromOthers.update(_.filterNot(_ == cb.baseHash))
+        } yield ()
+
+        acceptance.recoverWith {
+          case ex: PendingAcceptance =>
+            acceptErrorHandler(ex)
+          case error =>
+            pendingAcceptanceFromOthers.update(_.filterNot(_ == cb.baseHash)) *> acceptErrorHandler(error)
+        }
+
+        acceptance
+      case (NodeState.DownloadCompleteAwaitingFinalSync, Some(_)) =>
+        LiftIO[F].liftIO(dao.snapshotService.syncBufferAccept(checkpoint.checkpointCacheData))
```

Comment:
please pass `snapshotService` and here it will be `F`, not `IO` then

Reply:
Let's leave it for now it's out of scope of this PR, and the branch is growing to much anyway